pkg/netdevice/pciNetDevice_test.go

```diff
@@ -121,6 +124,133 @@ var _ = Describe("PciNetDevice", func() {
 				Expect(err).NotTo(HaveOccurred())
 			})
 		})
+		Context("with two devices but only one of them being RDMA", func() {
+			rc := &types.ResourceConfig{
+				ResourceName:   "fake",
+				ResourcePrefix: "fake",
+				SelectorObj: &types.NetDeviceSelectors{
+					IsRdma: true,
+				},
+			}
+			fs := &utils.FakeFilesystem{
+				Dirs: []string{
+					"sys/bus/pci/devices/0000:00:00.1/net/eth0",
+					"sys/bus/pci/devices/0000:00:00.2/net/eth1",
+					"sys/kernel/iommu_groups/0",
+					"sys/kernel/iommu_groups/1",
+					"sys/bus/pci/drivers/vfio-pci",
+				},
+				Symlinks: map[string]string{
+					"sys/bus/pci/devices/0000:00:00.1/iommu_group": "../../../../kernel/iommu_groups/0",
+					"sys/bus/pci/devices/0000:00:00.2/iommu_group": "../../../../kernel/iommu_groups/1",
+					"sys/bus/pci/devices/0000:00:00.1/driver":      "../../../../bus/pci/drivers/vfio-pci",
+					"sys/bus/pci/devices/0000:00:00.2/driver":      "../../../../bus/pci/drivers/vfio-pci",
+				},
+				Files: map[string][]byte{
+					"sys/bus/pci/devices/0000:00:00.1/numa_node": []byte("0"),
+					"sys/bus/pci/devices/0000:00:00.2/numa_node": []byte("0"),
+				},
+			}
+			defer fs.Use()()
+			defer utils.UseFakeLinks()()
+
+			rdma1 := &mocks.RdmaSpec{}
+			// fake1 will have 2 RDMA device specs
+			fake1ds := []*pluginapi.DeviceSpec{
+				&pluginapi.DeviceSpec{ContainerPath: "/fake/path", HostPath: "/dev/fake1a"},
+				&pluginapi.DeviceSpec{ContainerPath: "/fake/path", HostPath: "/dev/fake1b"},
+			}
+			rdma1.On("IsRdma").Return(true).On("GetRdmaDeviceSpec").Return(fake1ds)
+			// fake2 will have 0 rdma device specs to trigger error msg
+			rdma2 := &mocks.RdmaSpec{}
+			fake2ds := []*pluginapi.DeviceSpec{
+				&pluginapi.DeviceSpec{ContainerPath: "/fake/path", HostPath: "/dev/fake2"},
+			}
+			rdma2.On("IsRdma").Return(false).On("GetRdmaDeviceSpec").Return(fake2ds)
+
+			f := &mocks.ResourceFactory{}
+			f.On("GetDefaultInfoProvider", AnythingOfType("string"), "vfio-pci").Return(func(addr string, driver string) types.DeviceInfoProvider {
```

Comment:
I think it would be cleaner to create a mock.DeviceInfoProvider{} and use it here instead of using `vfio-pci`
as we dont really care in this test (and below test) for the default info provider right ?

you can then remove vfio-pci sysfs related stuff and simplify (abit) the test. WDYT ?

Reply:
I agree, we don't care of what they return. We do, however, care that they are called. Right?

Reply:
Updated: Used a mock and checked it's being called
WDYT?

Reply:
Perfect !